### PR TITLE
Add extension methods for using routing with Middleware

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

**Comment by davidfowl** [comment]
So happy

File: NuGetPackageVerifier.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
         ],
         "packages": {
             "Microsoft.AspNet.Routing.Abstractions": { },
+            "Microsoft.AspNet.Routing.Extensions": { }
```
Comment (JamesNK) [inline NuGetPackageVerifier.json]: Missing comma?

Comment (rynowak) [inline NuGetPackageVerifier.json]: Thanks :+1:

File: samples/RoutingSample.Web/project.json [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-    },
-    "frameworks": {
-        "dnx451": { },
-        "dnxcore50": { }
-    },
-    "webroot": "wwwroot"
+  "dependencies": {
+    "Microsoft.AspNet.Server.Kestrel": "1.0.0-*",
+    "Microsoft.AspNet.Server.WebListener": "1.0.0-*",
+    "Microsoft.AspNet.Routing": "1.0.0-*",
```
Comment (pranavkm) [inline samples/RoutingSample.Web/project.json]: You could remove this.

File: src/Microsoft.AspNet.Routing.Extensions/RequestDelegateRouteBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Routing;
+using Microsoft.AspNet.Routing.Constraints;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNet.Builder
+{
+    public static class RequestDelegateRouteBuilderExtensions
+    {
+        public static IRouteBuilder MapRoute(this IRouteBuilder builder, string template, RequestDelegate handler)
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing.Extensions/RequestDelegateRouteBuilderExtensions.cs]: Docs

File: src/Microsoft.AspNet.Routing.Extensions/RequestDelegateRouteBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            string verb,
+            string template,
+            Action<IApplicationBuilder> action)
+        {
+            var nested = builder.ApplicationBuilder.New();
+            action(nested);
+            return builder.MapVerb(verb, template, nested.Build());
+        }
+
+        public static IRouteBuilder MapVerb(
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing.Extensions/RequestDelegateRouteBuilderExtensions.cs]: Do we need an overload that accepts multiple method names?

File: src/Microsoft.AspNet.Routing.Extensions/project.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "warningsAsErrors": true,
+    "keyFile": "../../tools/Key.snk"
+  },
+  "dependencies": {
+    "Microsoft.AspNet.Routing": "1.0.0-*"
+  },
+  "frameworks": {
+    "net451": { },
+    "dotnet5.4": {
+      "dependencies": {
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing.Extensions/project.json]: remove

File: test/Microsoft.AspNet.Routing.Extensions.Tests/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿using System.Reflection;
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Routing.Extensions.Tests/Properties/AssemblyInfo.cs]: Shouldn't need this

File: test/Microsoft.AspNet.Routing.Tests/RouteTest.cs [hunk 7]
```diff
@@ -13,6 +13,7 @@
 using Microsoft.Extensions.Logging;
 using Moq;
 using Xunit;
+using Microsoft.Extensions.DependencyInjection;
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Routing.Tests/RouteTest.cs]: Sort usings.

File: test/Microsoft.AspNet.Routing.Tests/TemplateParserDefaultValuesTests.cs [hunk 8]
```diff
@@ -96,26 +96,28 @@ public void NonEmptyDefaultValue_WithOptionalParameter_Throws()
...
-
-            var serviceProviderMock = new Mock<IServiceProvider>();
-            serviceProviderMock.Setup(o => o.GetService(typeof(IInlineConstraintResolver)))
-                               .Returns(_inlineConstraintResolver);
-            routeBuilder.ServiceProvider = serviceProviderMock.Object;
+            var applicationBuilder = Mock.Of<IApplicationBuilder>();
+            applicationBuilder.ApplicationServices = services.BuildServiceProvider();
 
+            var routeBuilder = new RouteBuilder(applicationBuilder);
+            routeBuilder.DefaultHandler = new Mock<IRouter>().Object;
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Routing.Tests/TemplateParserDefaultValuesTests.cs]: `Mock.Of<IRouter>()`

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNet.Routing.Extensions/RequestDelegateRouteBuilderExtensions.cs (hunk 4 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Routing.Extensions/RequestDelegateRouteBuilderExtensions.cs]: I'd rather wait for user demand. All of this stuff can be written pretty easily.

File: src/Microsoft.AspNet.Routing.Extensions/project.json (hunk 5 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Routing.Extensions/project.json]: sure

**Pull request closed without merging**